Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -130,8 +130,66 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                         return deferMaybe(result)
                     }
                 }
-
                 return deferMaybe(result)
+            } >>== { result in
+                guard case .completed = result else {
+                    return deferMaybe(result)
+                }
+                return storage.getLocalBookmarksAdditions() >>== { newBookmarks -> Success in
+                    guard newBookmarks.count > 0 else {
+                        return succeed()
+                    }
+                    return buffer.getBufferItemWithGUID(BookmarkRoots.MobileFolderGUID).bind { maybeMobileRoot -> Deferred<Maybe<Record<BookmarkBasePayload>>> in
+                        // Update (or create!) the Mobile Root folder with its new children
+                        if maybeMobileRoot.isSuccess {
+                            let mobileRoot = maybeMobileRoot.successValue!
```

Comment:
Use `if let mobileRoot = maybeMobileRoot.successValue {` instead of this pattern.